shell/probe/raindrop.ps1

```diff
@@ -1,46 +1,39 @@
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
 function Run {
     Param([String]$Dat = "")
 
-    $TempFile = [System.IO.Path]::GetTempFileName() | Rename-Item -NewName { [System.IO.Path]::ChangeExtension($_, "exe") } -PassThru
-
+    $Vst = New-Item -Path "$Dir\$(New-Guid).exe" -Force
     $Headers = @{
-        'token' = $Token
+        'token' = FromEnv "PRELUDE_TOKEN"
         'dos' = $Dos
         'dat' = $Dat
     }
-    try {
-        $Response = Invoke-WebRequest -URI $Address -UseBasicParsing -Headers $Headers -MaximumRedirection 1 -OutFile $TempFile -PassThru
-    } catch {
-        $StatusCode = $_.Exception.Response.StatusCode.value__
-        Write-Output "ERROR: Failed to reach Prelude Service. " + $StatusCode
-        return
-    }
-    if ($CA -and $CA -ne $Response.BaseResponse.ResponseUri.Authority) {
-        return
-    }
+    $Response = Invoke-WebRequest -URI $Api -UseBasicParsing -Headers $Headers -MaximumRedirection 1 -OutFile $Vst -PassThru
     $Test = $Response.BaseResponse.ResponseUri.AbsolutePath.Split("/")[-1].Split("_")[0]
     if (-not $Test) {
-        Write-Output "INFO: Done running tests"
         return
     }
-
-    $TestExit = Execute $TempFile
-    Start-Process -FilePath $TempFile -ArgumentList "clean" -Wait -NoNewWindow -PassThru
-
-    Remove-Item $TempFile -Force
-    Run -Dat $($Test + ":" + $TestExit)
+    if ($CA -ne $Response.BaseResponse.ResponseUri.Authority) {
+        Write-Output "[P] Bad authority: $Response.BaseResponse.ResponseUri.Authority"
+        exit 0
+    }
+    Write-Output "[P] Running $Test [$Vst]" 
+    $Code = Execute $Vst   
+    Run -Dat "${Test}:$Code"
 }
 
 function Execute { 
     Param([String]$File)
+
     try {
-        return (Start-Process -FilePath $File -Wait -NoNewWindow -PassThru).ExitCode
+        $Code = (Start-Process -FilePath $File -Wait -NoNewWindow -PassThru).ExitCode
+        return If (Test-Path $Vst) {$Code} Else {127}
```

Comment:
return If throws an error. Either separate the lines or it probably works if you make it a sub-expression

Reply:
How sure are you?

![Screenshot 2023-03-08 at 3 46 39 PM](https://user-images.githubusercontent.com/49954156/223846052-00acd330-76a5-44f3-ac39-c52efaaeb33b.png)

Reply:
you've just taken the return out, which is the problem part, and showed me the sub-expression works..

Have you run this code path?